### PR TITLE
Add VFX Class's

### PR DESCRIPTION
**Pull request by At0ms**
All the vfx class's i have reversed so far, I can provide the sigs aswell if requested.

### CODE REVIEW

File: misc/CVfx.hpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+class CVFXTrafficLightColor
+{
+public:
+    float red; //0x0000
+    float green; //0x0004
+    float blue; //0x0008
+}; //Size: 0x000C
+
+// Credit: At0ms/Andy
```
Comment (Yimura) [inline misc/CVfx.hpp:98]: Add this at the top

File: misc/CVfx.hpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    float green;          // 0x28A4
+    char pad_28A8[44];      // 0x28A8
+    float blue;              // 0x28D4
+    char pad_28D8[108];      // 0x28D8
+    float intensity;      // 0x2944
+    float falloffmax;      // 0x2948
+    char pad_294C[120];      // 0x294C
+    float fadingDistance; // 0x29C4
+    float fadeLength;      // 0x29C8
+    char pad_29CC[13952]; // 0x29CC
```
Comment (Yimura) [inline misc/CVfx.hpp:47]: Is the CVFXTrain actually this big?

**Review by Yimura — CHANGES_REQUESTED** [review]
Where did you get the CVfx name? Is it somewhere in the game dumps or from somewhere in the game files?

Could you follow the same naming scheme as the other classes in this repo? Your class members don't start with `m_` and don't use `snake_casing`.

**Comment by At0ms** [comment]
Hi

I will make the changes as requested.

And no the train class should not be that big i will redo that class.

**Comment by At0ms** [comment]
> Where did you get the CVfx name? Is it somewhere in the game dumps or from somewhere in the game files?
> 
> Could you follow the same naming scheme as the other classes in this repo? Your class members don't start with `m_` and don't use `snake_casing`.

I found all the stuff i needed to reverse the vfx class's in this file visualsettings.dat found in the game rpfs
vfx is short for visualsettings but if you would like me to rename it to visualsettings i can

**Comment by Yimura** [comment]
> > Where did you get the CVfx name? Is it somewhere in the game dumps or from somewhere in the game files?
> > Could you follow the same naming scheme as the other classes in this repo? Your class members don't start with `m_` and don't use `snake_casing`.
> 
> I found all the stuff i needed to reverse the vfx class's in this file visualsettings.dat found in the game rpfs vfx is short for visualsettings but if you would like me to rename it to visualsettings i can

Nah it's fine as is, I'd rather it be like that.

**Comment by At0ms** [comment]
Making this a draft fornow as i never finished it.

**Comment by yasmasdas** [comment]
I know this pr is old but may I have some signatures? @At0ms

**Comment by Yimura** [comment]
Stale PR

**Pull request closed without merging**